sphinx/builders/xml.py

```diff
@@ -71,6 +71,7 @@ def write_doc(self, docname: str, doctree: Node) -> None:
         # work around multiple string % tuple issues in docutils;
         # replace tuples in attribute values with lists
         doctree = doctree.deepcopy()
+        doctree.document.attributes["xmlns:py"] = "https://github.com/sphinx-doc/sphinx/issues/9413"
```

Comment:
The "py" namespace comes from the domains mark-up.
https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html

And we have many kinds of domains (ex. C, C++, etc.). And developers can add them via API. So it would be better to add it with such code:
```suggestion
        for domain in self.env.domains.values():
            xmlns = "xmlns:" + domain.name
            doctree.document.attributes[xmlns] = "https://URL/to/somewhere"
```